### PR TITLE
Add a dashboard component that displays RMM memory

### PR DESCRIPTION
**Pull request by pentschev**
This is an overhaul of https://github.com/dask/distributed/pull/5740 , primarily updating colors as requested in that PR's review, which adds a visualization that provides a bit more information than the current GPU memory utilization plot. It also shows the amount of GPU memory being managed by RMM, and in the case of a RMM pool, the amount of pool memory actually used:

Screenshot 2023-03-29 at 15 35 29

There are green bars with varying opacity representing from left to right: the amount of memory used from the RMM pool, the amount of free memory in the RMM pool and the amount of memory allocated outside of the RMM pool.

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/dashboard/components/rmm.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                        format_bytes(rmm_used_worker),
+                        format_bytes(rmm_total_worker),
+                        format_bytes(gpu_used_worker),
+                        format_bytes(gpu_total_worker),
+                    )
+                )
+
+            self.memory_figure.title.text = dedent(
+                """\
+                RMM Utilization: {} / {}\n
```
Comment (jacobtomlinson) [inline distributed/dashboard/components/rmm.py:174]: **Nit:** I'm not sure the extra blank line is necessary.

```suggestion
                RMM Utilization: {} / {}
```

**Review by jacobtomlinson — APPROVED** [review]
Thanks for pushing this forward @pentschev. This looks good to me.

**Comment by jacobtomlinson** [comment]
I hope you don't mind @pentschev but I've pushed a commit to fix up the linting and tweak an extra line. I'll leave this to sit for a bit in case other folks want to weigh in, but otherwise, I intend to merge this tomorrow.

**Comment by pentschev** [comment]
> I hope you don't mind @pentschev but I've pushed a commit to fix up the linting and tweak an extra line. I'll leave this to sit for a bit in case other folks want to weigh in, but otherwise, I intend to merge this tomorrow.

Sure thing, thank Jacob!

**Comment by jacobtomlinson** [comment]
Test failures appear unrelated. Merging.

**Comment by pentschev** [comment]
Thanks @jacobtomlinson !

**Pull request merged**